apis/capabilities/v1beta1/product_types.go

```diff
@@ -1191,14 +1190,6 @@ func (product *Product) SetDefaults(logger logr.Logger) bool {
 		updated = true
 	}
 
-	// 3scale API ignores case of the system name field
```

Comment:
By default, the system name is taken from the info.title field in the OpenAPI definition (if it is not present in openApiCR).
It's stated in the documentation and in the code, and there - the conversion to lowercase is produced.
https://github.com/3scale/3scale-operator/blob/master/doc/openapi-user-guide.md#product-name
https://github.com/3scale/3scale-operator/blob/master/pkg/helper/openapi_utils.go#L25-L28

Suppose it was the reason that in product_types the conversion was done in very initial versions.

Maybe we could try find another way for fix, as for example for product, maybe do change here (?) - https://github.com/3scale/3scale-operator/blob/master/controllers/capabilities/product_threescale_reconciler.go#L70-L78 . For backend seems it's differ.   Thank you

Reply:
Approach to me seems good - however, needed to update this line as well https://github.com/3scale/3scale-operator/blob/master/pkg/helper/openapi_utils.go#L25-L28
Thanks!